e2e/core/core.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+module.exports = {
+    page: null,
+    setActivePageTo: function(pageName) {
+        try {
+            return this.page = require('../pages/'+ pageName)
+        } catch (err) {
+            this.fatalError('The page: ' + pageName + ' does not exist.\n error: ' + err);
+        }
+    },
+    goToPage: function(pageName) {
+        var page = this.setActivePageTo(pageName);
+        browser.ignoreSynchronization = page.ignoreSync;
+        return browser.get(page.url);
+    },
+    at: function(pageName){
+        var page = this.setActivePageTo(pageName);
+        return Promise.resolve(page.at()).then(function(v) {
```

Comment:
the at() function on a page is a statement that checks something that is unique to the page. example is in workspace.js. Because protractor returns a promise when asking for an element, it needs to be resolved before a meaningful error can be thrown.

Is there a better way to do this?